src/lineup/internal/cmds/filter.ts

```diff
@@ -173,3 +175,41 @@ export function restoreLineUpFilter(filter: LineUpStringFilterValue | IRegExpFil
 
   throw new Error('Unknown LineUp filter format. Unable to restore the given filter.');
 }
+
+/**
+ * String column's group criterias as stored in the provenance graph.
+ */
+interface ISerializedStringGroupCriteria extends IStringGroupCriteria {
+  values: string[];
+}
+
+/**
+ * Serializes LineUp StringColumn's `Group By` dialog's values, which can contain a RegExp objects to a string.
+ * The return value of this function can be passed to `JSON.stringify()` and stored in the provenance graph.
+ * @param groupBy Value returned from the `Group By` dialog
+ */
+export function serializeGroupByValue(groupBy: IStringGroupCriteria) {
+  const {type, values} = groupBy;
+  if (type === 'regex') {
+    return {
+      type,
+      values: values.map((value) => value.toString())
+    };
+  }
+  return groupBy;
+}
+
+/**
+ * Restores LineUp StringColumn's Group By` dialog's values from the provenance graph and returns an `IStringGroupCriteria`.
+ * @param groupBy Value as saved in the provenance graph.
+ */
+export function restoreGroupByValue(groupBy: ISerializedStringGroupCriteria) {
```

Comment:
Could you please add some unit tests for this function?

Reply:
@thinkh I added test for the serialize and restore functions. You can take a look again when you have time.